src/vs/workbench/services/userDataProfile/browser/userDataProfileImportExportService.ts

```diff
@@ -544,21 +543,41 @@ export class UserDataProfileImportExportService extends Disposable implements IU
 				return this.userDataProfilesService.createNamedProfile(`${profileTemplate.name} ${this.getProfileNameIndex(profileTemplate.name)}`, { shortName: profileTemplate.shortName, transient: temp });
 			}
 
-			const result = await this.dialogService.show(
-				Severity.Info,
-				localize('profile already exists', "Profile with name '{0}' already exists. Do you want to overwrite it?", profileTemplate.name),
-				[localize('overwrite', "Overwrite"), localize('create new', "Create New Profile"), localize('cancel', "Cancel")],
-				{ cancelId: 2 }
-			);
-			switch (result.choice) {
-				case 0: return profile;
-				case 2: return undefined;
+			enum ImportProfileChoice {
+				Overwrite = 0,
+				CreateNew = 1,
+				Cancel = 2
+			}
+			const { result } = await this.dialogService.prompt<ImportProfileChoice>({
```

Comment:
@sandy081 fyi please check, this one is a bit more complex adoption of the new dialog api